models/vrp.rb

```diff
@@ -407,15 +407,26 @@ def self.deduce_unavailable_days(hash, element, start_index, end_index, type)
       collect_unavaible_day_indices(element, start_index, end_index)
     end
 
-    def self.generate_schedule_indices_from_date(hash)
-      return hash if !hash[:configuration] || !hash[:configuration][:schedule]
-
-      if hash[:configuration][:schedule][:range_indices] &&
-         (hash[:services]&.any?{ |s| s[:unavailable_date_ranges] || s[:unavailable_visit_day_date] })
+    def self.detect_date_indices_inconsistency(hash)
+      missions = hash[:services].to_a + hash[:shipments].to_a
+      services_with_date = missions.any?{ |m| m[:unavailable_date_ranges] || m[:unavailable_work_date] }
+      services_with_index = missions.any?{ |m| m[:unavailable_index_ranges] || m[:unavailable_work_day_indices] }
+      vehicles_with_date = hash[:vehicles]&.any?{ |v| v[:unavailable_work_date] || v[:unavailable_date_ranges] }
+      vehicles_with_index = hash[:vehicles]&.any?{ |v| v[:unavailable_work_day_indices] || v[:unavailable_index_ranges] }
+      if (hash[:configuration][:schedule][:range_indices] &&
+         (services_with_date || vehicles_with_date)) ||
+         (hash[:configuration][:schedule][:range_date] &&
+         (services_with_index || vehicles_with_index))
```

Comment:
Maybe we can simplify the logic as follows:

```suggestion
      missions_and_vehicles = hash[:services].to_a + hash[:shipments].to_a + hash[:vehicles].to_a
      has_date = missions_and_vehicles.any?{ |m| m[:unavailable_date_ranges] || m[:unavailable_work_date] }
      has_index = missions_and_vehicles.any?{ |m| m[:unavailable_index_ranges] || m[:unavailable_work_day_indices] }
      if (hash[:configuration][:schedule][:range_indices] && has_date) ||
         (hash[:configuration][:schedule][:range_date] && has_index)
```